### PR TITLE
JR2-244 Clean CandidateService and implement candidate search

### PR DESCRIPTION
**Pull request by varusakov**

### CODE REVIEW

File: src/main/java/ch/admin/seco/service/candidate/config/LifecycleConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    private final ElasticsearchIndexService elasticsearchIndexService;
+
+    public LifecycleConfiguration(ElasticsearchIndexService elasticsearchIndexService) {
+        this.elasticsearchIndexService = elasticsearchIndexService;
+    }
+
+    @EventListener
+    public void handleContextRefresh(ApplicationReadyEvent event) {
+        elasticsearchIndexService.reindex();
+    }
```
Comment (michael-wirth) [inline src/main/java/ch/admin/seco/service/candidate/config/LifecycleConfiguration.java]: We use that mechanism to reindex the static data loaded by liquibase in the reference service.
But we shouldn't use it to index the candidates at every startup of the application. we don't do this for the jobs either, but there we also don't have any dummy data.

Remove this class.

If test require to have the data loaded into the elasticsearch then we will do it in the Test configuration.
in the dev environment we ca start the index via the swagger UI

File: src/main/java/ch/admin/seco/service/candidate/domain/search/CandidateProfileSynonym.java [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+import ch.admin.seco.service.candidate.domain.enums.Availability;
+import ch.admin.seco.service.candidate.domain.enums.DrivingCategory;
+import ch.admin.seco.service.candidate.domain.enums.Gender;
+import ch.admin.seco.service.candidate.domain.enums.WorkForm;
+
+@Document(indexName = "candidate-profiles", type = "candidateProfile")
+@Mapping(mappingPath = "config/elasticsearch/mappings/candidate-profile.json")
+@Setting(settingPath = "config/elasticsearch/settings/folding-analyzer.json")
+public class CandidateProfileSynonym {
```
Comment (michael-wirth) [inline src/main/java/ch/admin/seco/service/candidate/domain/search/CandidateProfileSynonym.java]: this is not a `Synonym` it's just the ElasticSearch representation of the CandidateProfile

File: src/main/java/ch/admin/seco/service/candidate/domain/search/JobExperienceSynonym.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
+package ch.admin.seco.service.candidate.domain.search;
+
+import ch.admin.seco.service.candidate.domain.enums.Degree;
+import ch.admin.seco.service.candidate.domain.enums.Graduation;
+
+public class JobExperienceSynonym {
```
Comment (michael-wirth) [inline src/main/java/ch/admin/seco/service/candidate/domain/search/JobExperienceSynonym.java]: Same. This is not a synonym. Remove the postfix

File: src/main/java/ch/admin/seco/service/candidate/domain/search/LanguageSkillSynonym.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
+package ch.admin.seco.service.candidate.domain.search;
+
+public class LanguageSkillSynonym {
```
Comment (michael-wirth) [inline src/main/java/ch/admin/seco/service/candidate/domain/search/LanguageSkillSynonym.java]: here as well

File: src/main/java/ch/admin/seco/service/candidate/service/impl/CandidateServiceImpl.java [hunk 5]
```diff
@@ -77,25 +102,140 @@ public CandidateProtectedData findOne(UUID id) {
...
-     */
+
     @Override
-    @Transactional(readOnly = true)
-    public Page<CandidateProtectedData> search(String query, Pageable pageable) {
-        log.debug("Request to search for a page of Candidates for query {}", query);
-//        Page<CandidateProtectedData> result = candidateSearchRepository.search(queryStringQuery(query), pageable);
-//        return result;
-        return null;
+    public CandidateProfile saveProfile(final CandidateProfile candidateProfile) {
```
Comment (michael-wirth) [inline src/main/java/ch/admin/seco/service/candidate/service/impl/CandidateServiceImpl.java]: rename to `save`. the parameter will make the difference

File: src/main/java/ch/admin/seco/service/candidate/domain/search/CandidateProfileSynonym.java (hunk 2 above)
Comment (varusakov) [reply src/main/java/ch/admin/seco/service/candidate/domain/search/CandidateProfileSynonym.java]: I think that we need some special postfix to indicate that classes are designed for ElasticSearch. How about `Document` postfix?

Comment (michael-wirth) [reply src/main/java/ch/admin/seco/service/candidate/domain/search/CandidateProfileSynonym.java]: ok, agree

**Pull request merged**